ci: add automated spell checking

dviererbe opened the pull request:
As discussed in the last MIR meeting. This PR adds the GitHub Action [`check-spelling`](https://github.com/marketplace/actions/check-spelling).

I already configured some terminology used in the canonical/ubuntu/MIR context.

Note: This spell checker is not perfect (e.g. normal words like `plugins`, `todo`, `architetcures` had to be added), but as a demonstration of its capabilities, it found these misspelled words #18.

dviererbe commented:
You can see an example spell-checker error report here: https://github.com/dviererbe/ubuntu-mir/commit/9ae06d89d6dec06edaf3e69a54e7ceb6e86ad1c7#commitcomment-114607322

dviererbe commented on .github/actions/spelling/allow/linux.txt:
```diff
@@ -0,0 +1,47 @@
+media
+mnt
+opt
+opt
```
Oh, I missed that. I will fix that

dviererbe replied:
Resolved with https://github.com/canonical/ubuntu-mir/pull/19/commits/ec04ed11e4d943d887e625ea30624902003a01e2

slyon commented on .github/actions/spelling/allow/ubuntu-misc.txt:
```diff
@@ -0,0 +1,20 @@
+restricted
+sru
+triaged
+uife
```
should this be "uefi"?

dviererbe replied:
Resolved with https://github.com/canonical/ubuntu-mir/pull/19/commits/3e7dd8342c82d456296ba8538f2192b63f6df9ee

dviererbe replied:
Thanks for catching this!

dviererbe replied:
Ohhh... I remember again that this was the acronym UI Freeze Exception 🤦🏻

setharnold commented on .github/actions/spelling/allow/formats.txt:
```diff
@@ -0,0 +1,70 @@
+basic
+c
+c#
+c++
```
I thought the tool should have stripped out the punctuation here, were these needed? (same with `f#` later)

jsoref replied:
The default minimum word size is `3` characters and in 0.0.21 the only characters it considers are `a-z` and `'` (with some aliases for this character).

dviererbe replied:
Resolved with https://github.com/canonical/ubuntu-mir/pull/19/commits/d7312d2433518149bbec0295d625b2a67fb767e3

I added this to `.github/actions/spelling/allow/README.md` with https://github.com/canonical/ubuntu-mir/pull/19/commits/e8224fb04f45d8b89e5df77ba860b4b87658809b

dviererbe replied:
Thanks for noticing that!

setharnold commented on .github/actions/spelling/allow/formats.txt:
```diff
@@ -0,0 +1,70 @@
+markdown
+matlab
+object
+objective-c
```
oh yes, this too? is this needed?

setharnold reviewed (commented):
Looks awesome, thanks :D

eslerm commented:
this looks great!

the tool finds [*illegal patterns*](https://github.com/dviererbe/ubuntu-mir/actions/runs/5057531705/jobs/9076410849#step:2:1) which may need to be resolved in a separate PR

jsoref commented:
You really want to fix those things before/in this PR otherwise people who fork from once this merges will be confused/unhappy.

dviererbe commented:
I updated the comments in `.github/actions/spelling/line_forbidden.patterns` to give users a quick action how to fix the error message.

I also added the error fixes to #18.

jsoref commented on .github/actions/spelling/line_forbidden.patterns:
```diff
@@ -1,61 +1,53 @@
 \bGithub\b
 
-# s.b. GitLab
+# If you encounter this error, use spelling "GitLap" instead.
```
```suggestion
# If you encounter this error, use spelling "GitLab" instead.
```

dviererbe replied:
I updated the last commit, thanks!
